pepper-apis/src/test/java/org/broadinstitute/ddp/util/TestingUserUtil.java

```diff
@@ -147,16 +147,19 @@ public static void deleteTestUser(
             throws Auth0Exception {
 
         Auth0MgmtTokenHelper mgmtTokenHelper = Auth0Util.getManagementTokenHelperForStudy(handle, studyGuid);
+        UserDto user = handle.attach(JdbiUser.class).findByUserGuid(userGUID);
         CachedUser cachedUser = tryCachedUser(userGUID, auth0ClientId, mgmtTokenHelper.getDomain());
         if (cachedUser != null) {
-            LOG.info("Using cached test user");
-            return cachedUser.asTestingUser();
+            if (cachedUser.getId() == user.getUserId()) {
```

Comment:
I noticed that sometimes when you switch from local mysql to testcontainers, it messes with the cached user token because the cached data no longer matches up to what's in the new database, so I added this check to discard the cached data.